pdfparser/poppler.pyx

```diff
@@ -575,134 +577,211 @@ cdef class LinesIterator:
-            assert len(words[-1]) == word_length
+    def __cinit__(self, PopplerPage page):
+        self.page = page
+        self.word = NULL
+
+    cdef Word wrap(self, TextWord *word):
+        cdef:
+            GooString *_text
+            GooString *_font_name
+            int        i, word_length
+            double     l, t, r, b, g
+
+        self.char_bboxes = []
+        self.char_fonts = CompactList()
 
-            # update the line bounding box
-            word.getBBox(&bx1, &by1, &bx2, &by2)
+        word_length = word.getLength()
 
-            if bx1 < self.x1 or self.x1 == 0:
-                self.x1 = bx1
+        # gets bounding boxes for all characters and font info
+        for i in range(word_length):
+            word.getCharBBox(i, &l, &t, &r, &b)
+            self.char_bboxes.append(BBox(l, t, r, b))
 
-            if by1 < self.y1 or self.y1 == 0:
-                self.y1 = by1
+            _font_name = word.getFontName(i)
 
-            if bx2 > self.x2:
-                self.x2 = bx2
+            # non-embedded fonts have `_font_name` set to NULL.
+            if _font_name != NULL:
+                try:
+                    font_name = _font_name.getCString().decode('UTF-8')
+                except:
+                    font_name = u'%r' % _font_name.getCString()
+            else:
+                font_name = u'?'
 
-            if by2 > self.y2:
-                self.y2 = by2
+            word.getColor(&r, &g, &b)
+            self.char_fonts.append(FontInfo(font_name,
+                                            word.getFontSize(),
+                                            Color(r, g, b)))
 
-            # add space after word if necessary
-            if word.hasSpaceAfter():
-                words.append(u' ')
-                self._bboxes.append(BBox(last_bbox.x2, last_bbox.y1, last_bbox.x2, last_bbox.y2))
-                self._fonts.append(last_font)
+        # store the word's content
+        _text = word.getText()
+        self.text = _text.getCString().decode('UTF-8')
+        del _text
 
-            # continue with the next word in the line
-            word = word.getNext()
+        # store store the word's bounding box
+        word.getBBox(&l, &t, &r, &b)
+        self.bbox = BBox(l, t, r, b)
 
-        self._text= u''.join(words)
+        # store `has space after` flag
+        self.has_space_after = word.hasSpaceAfter()
 
-        # the number of bboxes/fonts must equal the number of characters in the word
-        assert len(self._bboxes) == len(self._text), 'number of bounding boxes do not match number of charecters: %d != %d' % (len(self._bboxes), len(self._text))
+        # sanity check for data integrity
+        assert len(self.text) == len(self.char_bboxes)
+        assert len(self.text) == len(self.char_fonts)
+
+        return self
 
     property bbox:
-        '''
-        The bounding box of this line of text
-        as (left, top, right, bottom).
```

Comment:
Would be good to move to the `classs BBox` so that we know the x/y convention.

Reply:
I added this comment there as you suggested.